### PR TITLE
Drop support for Plone 5.2 and for Python 3.7 and lower

### PR DESCRIPTION
**Pull request by mauritsvanrees**

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
This is ready for review now.  
Strangely the 'Changelog verifier' test fails.  This PR adds multiple snippets in the news directory, and they are correctly named, so should be fine.

Note that this package is only in the `sources.cfg` of coredev 6.0, not 6.1, so only 6.0 needs to be tested.

@jenkins-plone-org please run jobs

**Comment by mauritsvanrees** [comment]
All robot tests fail, which happens often these days.

And one line from a CMFPlone test needs to be [removed](https://github.com/plone/Products.CMFPlone/pull/3998/commits/654c1e05612e154be1d791456bb3bb5703719bc3) which I do as part of [this CMFPlone PR](https://github.com/plone/Products.CMFPlone/pull/3998).  Let's first get that one merged separately.

**Comment by mauritsvanrees** [comment]
The 'Changelog verifier' check can be ignored.  As Steve found out, this is due to a new towncrier release.  Should be fixed with the next towncrier release.

@jenkins-plone-org please run jobs

File: CHANGES.rst [hunk 1]
```diff
@@ -161,7 +161,7 @@ Bug fixes:
   Archetypes 1.4.2 (used for Plone 2.5.2) of PortalTransforms is then the 1.5
   final.
   We dont need increasing of release numbers because of Plone 3.0,
-  Archetypes 1.5, ... if theres no change in the dependend product, like
+  Archetypes 1.5, ... if there's no change in the dependent product, like
```
Comment (jensens) [inline CHANGES.rst:164]: 🥇  LOL yeah, a change log by me from 2006 got corrected

**Review by jensens — APPROVED** [review]
this is wild 👍🏼

**Pull request merged**